### ISSUE TITLE
Queen-Attack should have unit test to check if queens on same square

### ISSUE THREAD

**Issue by srpeterson** [exercism/exercism#5195]
Hi - I'm going through the F# track. In the Queen-Attack module I think there should be a unit test to check and see if 'canAttack' returns false if queen1 and queen2 are on the same square:

[<Fact>]
let ``Queens cannot attack if on same square`` () =
    let whiteQueen = (4, 4)
    let blackQueen = (4, 4)
    canAttack blackQueen whiteQueen |> should equal false

'canAttack' cannot assume that it will be passed two different values for queen1 and queen2. In real life two pieces cannot occupy the same square at the same time, so 'canAttack' should take into account this edge case, in my humble opinion!

**Comment by iHiD** [comment]
cc @exercism/fsharp

**Comment by srpeterson** [comment]
Updated md in original comment so now shows full unit test function

**Comment by NobbZ** [comment]
Its currently not covered by the canonical data:

https://github.com/exercism/problem-specifications/blob/master/exercises/queen-attack/canonical-data.json

So it should be added there as well.

Though I'd not say, that the function should return `false` in this case, but instead signal an error for invalid input in the lannguages idioms.

On the other hand side, that would overcomplicate the exercise… I think we should just assume for this exercise, that such an invalid game state is guaranteed to not happen due to the surrounding game implementation.

**Comment by srpeterson** [comment]
I don't think it would over complicate the exercise. Having the queen1 = queen2 check (whether it signals an error or returns false) is easy to implement. I did it in my solution (even though there is no unit tests for it currently). In my opinion, it helps the developer learn to consider all edge cases.

See:  [https://exercism.io/tracks/fsharp/exercises/queen-attack/solutions/7e229407a0424297b5dc3d27cd6a1c9d](https://exercism.io/tracks/fsharp/exercises/queen-attack/solutions/7e229407a0424297b5dc3d27cd6a1c9d)

Just a suggestion, my F# Exercism friends!
